backend/FwLite/FwLiteShared/Services/ProjectServicesProvider.cs

```diff
@@ -38,22 +38,30 @@ public async Task DisposeService(DotNetObjectReference<IAsyncDisposable> service
     }
 
     [JSInvokable]
-    public async Task<ProjectScope> OpenCrdtProject(string projectName)
+    public Task<ProjectData> GetCrdtProjectData(Guid projectId)
+    {
+        var crdtProject = crdtProjectsService.GetProject(projectId)
+            ?? throw new InvalidOperationException($"Crdt Project {projectId} not found");
+        return Task.FromResult(crdtProject.Data ?? throw new InvalidOperationException($"Project data for {crdtProject.Name} not found"));
+    }
+
+    [JSInvokable]
+    public async Task<ProjectScope> OpenCrdtProject(Guid projectId)
     {
         var serviceScope = serviceProvider.CreateAsyncScope();
         var scopedServices = serviceScope.ServiceProvider;
-        var project = crdtProjectsService.GetProject(projectName) ??
-                      throw new InvalidOperationException($"Crdt Project {projectName} not found");
+        var project = crdtProjectsService.GetProject(projectId)
+            ?? throw new InvalidOperationException($"Crdt Project {projectId} not found");
```

Comment:
In looking at this error, and thinking about what it would take to debug it, which would be non trivial. I'm starting to question what the actual tangible benefit is of using project ID here and in a number of other places where we have to throw an error like this. Maybe we should just be using the file name so that when it's missing we can throw a helpful error with the name of the project.

Reply:
As discussed, we'll revert this and use the code as the unique identifier on the client.